### PR TITLE
#161967126 User email verification upon registration

### PR DESCRIPTION
**Pull request by Bryan-Cee**
**What does this PR do?**

Enable email verification upon registration.

**Description of tasks to be completed**

- Add email verification upon registration.
- Fixed errors raised by Flake8
- Add tests for email verification
- Refactor failing test

**How should this be manually tested?**

Clone the repo from your terminal: `$ git clone https://github.com/andela/ah-code-titans.git
`
Create a virtual environment: `$ python3 -m venv <name_of_environment>`

Activate the virtual environment: `$ source <name_of_environment>/bin/activate`

Install project dependencies: `$ pip install -r requirements.txt`

Add the following environment variables and populate them with your settings:

-`$ export DB_NAME='choose_a_dbname'`
-`$ export DB_USER='db_username'`
-`$ export DB_PASSWORD='db_role_password`'
-`$ export DB_HOST='localhost or preffered host'`
-`$ export DB_PORT='server_port_number'`
-`$ export SECRET_KEY='your_secret_key'`

Switch to the appropriate branch: `$ git checkout ft-registration-email-verification-161967126`
Run migrations: `$ python manage.py migrate`
Run tests: `$ python manage.py test`

What are the relevant Pivotal Tracker stories?
[#161967126](https://www.pivotaltracker.com/story/show/161967126)

Checklist

- [x]   My code follows the style guidelines of this project 
- [x]   I have performed a self-review of my own code
- [x]   I have commented my code, particularly in hard-to-understand areas
- [x]   I have made corresponding changes to the documentation
- [x]   My changes generate no new warnings

Screenshots
screenshot 2018-12-06 at 10 04 19
screenshot 2018-12-07 at 11 36 34

### CODE REVIEW

**Review by jillo-abdullahi — CHANGES_REQUESTED** [review]
`test_login_with_inactive_account` is failing locally. There could be an env variable you have not added here.

**Comment by jillo-abdullahi** [comment]
@Bryan-Cee You can rebase to develop to pick up @davishooly's changes and try re-running your tests.

**Review by jillo-abdullahi — APPROVED** [review]

**Review by davishooly — APPROVED** [review]

File: authors/apps/authentication/tests/test_login.py [hunk 1]
```diff
@@ -59,15 +59,14 @@ def test_login_with_empty_password_field(self):
             "This field may not be blank."
         )
 
-    def test_login(self):
+    def test_login_with_inactive_account(self):
         """ test login success """
```
Comment (denomash) [inline authors/apps/authentication/tests/test_login.py]: update this comment to match the function of the test

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by denomash — APPROVED** [review]
👍

File: authors/apps/authentication/tests/test_register.py [hunk 2]
```diff
@@ -16,6 +19,27 @@ def register_user(self, data):
             content_type='application/json'
         )
 
+    def test_registration_email_verification(self):
+
+        response_details = self.register_user(self.new_user)
+        user_details = User.objects.get(username=self.new_user['user']['username'])
+        from django.utils.encoding import force_bytes
```
Comment (bettblake08) [inline authors/apps/authentication/tests/test_register.py]: Please move the import to the top.

**Review by bettblake08 — CHANGES_REQUESTED** [review]

**Review by mutoromike — CHANGES_REQUESTED** [review]
Kindly add the register screenshot.

**Review by bettblake08 — COMMENTED** [review]
Response message should inform the user check their email.

**Review by bettblake08 — APPROVED** [review]
👍 Looks good!

**Review by mutoromike — APPROVED** [review]

**Pull request merged**